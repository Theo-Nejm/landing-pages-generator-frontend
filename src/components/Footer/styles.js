import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    p,
    a,
    i,
    b {
      font-size: ${theme.fonts.sizes.xxsmall};
      text-align: center;
    }
  `}
`;
