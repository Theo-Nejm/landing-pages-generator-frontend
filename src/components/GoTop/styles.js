import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    color: #fff;
    background: ${theme.colors.primaryColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    border-radius: 50%;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    z-index: 4;
    opacity: 0.85;
  `}
`;
