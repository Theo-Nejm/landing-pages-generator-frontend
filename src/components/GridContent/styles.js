import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
  `}
`;

export const HTMLContainer = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xlarge} 0;
  `}
`;
