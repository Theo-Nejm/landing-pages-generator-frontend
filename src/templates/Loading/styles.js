import styled, { keyframes } from 'styled-components';

const rotate = () => keyframes`
  0% {
    transform: translate(-50%, -50%) rotate(0deg);
    border-left: 0.5rem solid #dc143c;
    border-top: 0.5rem solid #dc143c;
  }

  25% {
    transform: translate(-50%, -50%) rotate(90deg);
    border-left: 0.5rem solid #FE365E;
    border-top: 0.5rem solid #FE365E;
  }

  50% {
    transform: translate(-50%, -50%) rotate(180deg);
    border-left: 0.5rem solid #FFD9D9;
    border-top: 0.5rem solid #FFD9D9;
  }

  75% {
    transform: translate(-50%, -50%) rotate(270deg);
    border-left: 0.5rem solid #FE365E;
    border-top: 0.5rem solid #FE365E;
  }

  100% {
    transform: translate(-50%, -50%) rotate(360deg);
    border-left: 0.5rem solid #dc143c;
    border-top: 0.5rem solid #dc143c;
  }
`;

export const Container = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;

  background: #0a1128;
  z-index: 10;

  &:before,
  &:after {
    content: '';
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);

    border: 0.5rem solid transparent;
    border-radius: 50%;
  }

  &:after {
    content: '';
    width: 6rem;
    height: 6rem;
    border-left: 0.5rem solid #dc143c;
    border-top: 0.5rem solid #dc143c;
    animation: ${rotate()} 2s linear infinite;
  }

  &:before {
    content: '';
    width: 20px;
    height: 20px;
    border-left: 0.5rem solid #dc143c;
    border-top: 0.5rem solid #dc143c;
    animation: ${rotate()} 1s linear infinite reverse;
  }
`;
