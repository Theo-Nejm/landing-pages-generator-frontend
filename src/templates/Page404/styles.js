import styled from 'styled-components';

export const Container = styled.div`
  height: 100vh;
  width: 100vw;
  background: #F0F0F0;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #0a1128;

  > div {
    width: 60rem;
    height: max-content;
    background: #FFF;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.15);
    border-radius: 2rem;
    padding: 3rem;
    display: flex;
    flex-direction: column;
    gap: 3.5rem;
    align-items: center;
    justify-content: center;
    text-align: center;

    span {
      font-size: 8rem;

      i {
        color: #dc143c;
        font-style: normal;
      }
    }

    a {
      height: 3rem;
      font-size: 2.5rem;
      font-weight: 600;
    }
  }
}
`;
