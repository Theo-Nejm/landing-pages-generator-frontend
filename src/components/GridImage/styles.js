import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    padding: 4rem 0;
    width: 80vw;

    h2 {
      font-size: 4.2rem;
    }

    ${TextComponent} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const Element = styled.div`
  ${() => css`
    overflow: hidden;

    > img {
      width: 100%;
      transition: all 0.3s ease-in-out;

      &:hover {
        transform: scale(1.2) rotate(10deg);
      }
    }
  `}
`;
