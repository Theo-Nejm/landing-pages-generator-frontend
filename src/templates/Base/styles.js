import styled, { css } from 'styled-components';
import { Container as TextContainer } from '../../components/TextComponent/styles';

export const Container = styled.div`
  ${() => css`
    ${TextContainer} {
      font-size: 2rem;
    }

    & > div:nth-child(1) {
      padding-top: 7rem;
    }
  `}
`;
