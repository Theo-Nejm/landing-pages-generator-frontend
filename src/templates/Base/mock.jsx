import linksMock from '../../components/NavBar/mock';
import gridTextMock from '../../components/GridText/mock';
import gridImageMock from '../../components/GridImage/mock';
import { GridText } from '../../components/GridText';
import { GridImage } from '../../components/GridImage';

export const templateMock = {
  children: (
    <div>
      <GridText {...gridTextMock} />
      <GridImage {...gridImageMock} background />
      <GridText {...gridTextMock} />
      <GridImage {...gridImageMock} background />
      <GridText {...gridTextMock} />
    </div>
  ),
  links: linksMock,
  footerHtml: '<i>Made with <3 by <b>Theo Nejm</b></i>',
  logoData: {
    text: 'Minha logo',
    srcImg: 'assets/images/logo-lp.svg',
    link: '#',
  },
};
