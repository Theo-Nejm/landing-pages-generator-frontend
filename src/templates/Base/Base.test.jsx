import { Base } from '.';
import { renderTheme } from '../../styles/renderTheme';
import { templateMock } from './mock';

describe('<Base />', () => {
  it('should render <Base />', () => {
    const { conatiner } = renderTheme(<Base {...templateMock} />);
    expect(conatiner).toMatchSnapshot();
  });
});
