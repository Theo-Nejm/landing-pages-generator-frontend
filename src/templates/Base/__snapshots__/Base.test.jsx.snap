// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should render <Base /> 1`] = `undefined`;
